espnet2/bin/asr_inference.py

```diff
@@ -166,6 +174,49 @@ def __init__(
         self.dtype = dtype
         self.nbest = nbest
 
+    def from_huggingface(huggingface_id: str, **kwargs) -> "Speech2Text":
+        """Instantiate a Speech2Text model from a local packed archive or a model id
```

Comment:
Will you implement `from_huggingface` for all inference classes? This function is completely same for them, so why don't you implement a function to download  as https://github.com/espnet/espnet_model_zoo? It's better to avoid duplication.

For compatibility, I think it would be better to implement this function in espnet_model_zoo rather than in espnet module using same API.

Reply:
Yeah, I think that it could be better. @sw005320 what do you think about it?

Reply:
Note that `from_pretrained` is a popular name, which can be seen in other module, like fairseq and astreroid, so I think it's might be even better to define a function in model zoo API for backward compatibility and invoke it `Speech2Text.from_pretraind()` for userbility.

Reply:
Given the changes, I think that I will need to remove the part from the `inference` method and keep `from_pretrained` but this will call the download function from `espnet_model_zoo`. is this fine? I will commit the changes soon.

Reply:
> Given the changes, I think that I will need to remove the part from the `inference` method and keep 


I'm not sure. Do you mean removing the following part ?https://github.com/espnet/espnet/blob/ce412369a11eef160c0468b48ca5ef348bc7f33c/espnet2/bin/asr_inference.py#L330-L334

Do you need to remove it? I think this mode is useful.


> `from_pretrained` but this will call the download function from `espnet_model_zoo`. is this fine? I will commit the changes soon.

Yes, it's fine for me.